mathenger-api/src/main/java/com/example/mathengerapi/integrations/dspace/DSpaceMessageHandler.java

```diff
@@ -28,8 +30,20 @@ public void onNewMessage(MessageSent event) {
     private void handleBotMessage(MessageSent event) {
         // TODO refactor to support many different commands without bunch of "if" blocks
         var message = event.getText();
-        if ("/community".equals(message)) {
-            botCommandsHandler.handleAllCommunities(event.getChatId());
+        if (!event.getAuthorId().equals(botInfoHolder.getBotAccount().getId())) {
+            if ("/community".equals(message)) {
+                botCommandsHandler.handleAllCommunities(event.getChatId());
+            } else if (message.matches("/community_(?i)[a-f\\d]{8}-([a-f\\d]{4}-){3}[a-f\\d]{12}")) {
+                UUID communityId = UUID.fromString(message.substring(11, message.length()));
+                botCommandsHandler.handleAllCollectionsOfCommunity(event.getChatId(), communityId);
+            } else if (message.matches("/colpublications_(?i)[a-f\\d]{8}-([a-f\\d]{4}-){3}[a-f\\d]{12}")) {
+                UUID collectionId = UUID.fromString(message.substring(17, message.length()));
+                botCommandsHandler.handleAllItemsOfCollection(event.getChatId(), collectionId);
+            } else if ("/help".equals(message)) {
+                botCommandsHandler.handleAllCommands(event.getChatId());
+            } else {
+                botCommandsHandler.handleIfErrorCommand(event.getChatId(), message);
```

Comment:
Bot will respond to all messages in GROUP_CHAT. Check whether the chat type is PRIVATE before handling it.